test/unit/utils/TagUtils.ts

```diff
@@ -0,0 +1,141 @@
+import test from "ava";
+import TagUtils from '../../../src/utils/TagUtils';
+import _ from "lodash";
+import sinon, { SinonSandbox } from "sinon";
+import { TestEnvironment } from '../../support/sdk/TestEnvironment';
+import { TagCategory, TagsObjectForApi, TagsObjectWithBoolean, Categories } from '../../../src/models/Tags';
+import { deepCopy } from '../../../src/utils';
+
+const sinonSandbox: SinonSandbox = sinon.sandbox.create();
+
+test.beforeEach(async () => {
+    await TestEnvironment.initialize({ stubSetTimeout: true });
+    TestEnvironment.mockInternalOneSignal();
+});
+
+test.afterEach(() => {
+    sinonSandbox.restore();
+});
+
+test('check conversion from tag boolean values to number strings', t => {
+    const converted = TagUtils.convertTagsBooleansToApi({ tag1: false, tag2: true });
+    t.true(_.isEqual(converted, { tag1: "0", tag2: "1" }));
+});
+
+test('check conversion from tag string values to booleans', t => {
+    const converted = TagUtils.convertTagsApiToBooleans({ tag1: "1", tag2: "0" });
+    t.true(_.isEqual(converted, { tag1: true, tag2: false }));
+});
+
+test('check correct object difference is returned', t => {
+    let localTags: TagsObjectForApi = { tag1: "1", tag2: "0" };
+    let playerTags: TagsObjectForApi = { tag1: "0" };
+    const diff1 = TagUtils.getObjectDifference(localTags, playerTags);
+
+    localTags = { tag1: "0", tag2: "1" };
+    playerTags = { tag1: "1", tag2: "1" };
+    const diff2 = TagUtils.getObjectDifference(localTags, playerTags);
+
+    localTags = { tag1: "1" };
+    playerTags = { tag2: "1" };
+    const diff3 = TagUtils.getObjectDifference(localTags, playerTags);
+
+    localTags = { tag2: "1" };
+    playerTags = { tag2: "1" };
+    const diff4 = TagUtils.getObjectDifference(localTags, playerTags);
+
+    t.true(_.isEqual(diff1, { tag1: "1", tag2: "0" }));
```

Comment:
This line could be simplified with `t.is()` instead of using ` t.true(_.isEqual())`

Same with the below 3 lines plus other test below that.

Reply:
Done.